packages/graph/hash_graph/lib/graph/src/api/rest/data_type.rs

```diff
@@ -67,16 +69,14 @@ struct CreateDataTypeRequest {
     ),
     request_body = CreateDataTypeRequest,
 )]
-async fn create_data_type<S: Store>(
+async fn create_data_type<S: Store + Send + Sync>(
     body: Json<CreateDataTypeRequest>,
-    store: Extension<S>,
+    store: Extension<Arc<S>>,
 ) -> Result<Json<Persisted<DataType>>, StatusCode> {
-    let Json(body) = body;
-    let Extension(store) = store;
```

Comment:
How are we able to get rid of this destructuring? Does `Extension` impl `Deref` for it's inner generic?

Reply:
Yes, those are implementing `Deref`